app/main/authorisation.py

```diff
@@ -0,0 +1,31 @@
+import pandas as pd
+
+from app.const import ORGANISATION_ID_TO_PLACE_NAMES
+from config import Config
+
+
+def get_local_authority_place_names(user_email: str) -> None | list[str]:
+    """
+    Get the local authority place names corresponding to a user's email domain.
+
+    This function takes a user's email address and uses the domain part (after '@')
+    to look up the corresponding local authority from a CSV file. If a matching local
+    authority is found, its identifier is returned, and this is used to look up the
+    place names for which the local authority is expected to submit for.
+
+    :param user_email: A string representing the user's email address.
+    :return: An array of place names or None if none are found.
+    """
+    lookup_url = Config.LOCAL_AUTHORITIES_LOOKUP
+    df_lookup = pd.read_csv(lookup_url, index_col=0)
+
+    organisation_id = df_lookup[df_lookup["homepage_url"].str[11:-1] == user_email.split("@")[1]]["gss"]
+
+    # some urls do not end with "/"
+    if len(organisation_id) == 0:
+        organisation_id = df_lookup[df_lookup["homepage_url"].str[11:] == user_email.split("@")[1]]["gss"]
+
+    if len(organisation_id) == 0:
```

Comment:
Could we maybe use a regex instead to capture both styles? As is it makes the logic hard to parse? Or maybe use a library like https://pypi.org/project/tld/ ?